gcc-4.8/post-link.sh

```diff
@@ -0,0 +1,136 @@
+    target_os_md5=( `cat /etc/*-release | md5sum` )
+
+    # No need to make any portability fixes if 
+    # we're deploying to the same OS we built with.
+    if [[ "${build_os_md5[0]}" == "${target_os_md5}" ]]; then
+        echo "gcc install OS matches gcc build OS: Skipping post-link portability fixes."
+    else
+
+        # In this script, we attempt to fix 3 Linux distro portability issues:
+    
+        #
+        # Linux Portability Issue #1: "fixed includes"
+        #
+    
+        # Remove the headers that gcc "fixed" as part of the gcc build process.
+        # They kill the gcc binary's portability to other systems,
+        #   and shouldn't be necessary on ANSI-compliant systems anyway.
+        # More discussion can be found here:
+        # https://groups.google.com/a/continuum.io/d/msg/conda/HwUazgD-hJ0/aofO0vD-MhcJ
+        while read x ; do
+          grep -q 'It has been auto-edited by fixincludes from' "${x}" \
+                   && rm -f "${x}"
+        done < <(find ${PREFIX}/lib/gcc/*/*/include*/ -name '*.h')
+    
+        #
+        # Linux Portability Issue #2: linker needs to locate crtXXX.o
+        #
+    
+        # Locate the system's C-runtime object files and link them into the gcc 
+        #  build so they are automatically on the gcc search path.
+        # (The location of these files varies from one system to the next.)
+        C_RUNTIME_OBJ_FILES="crt0.o crt1.o crt2.o crt3.o crti.o crtn.o"    
+
+        for obj_file in $C_RUNTIME_OBJ_FILES; do
+            for libdir in `/sbin/ldconfig -v 2> /dev/null | grep '^/' | awk '{print $1}'`; do
+                # Remove trailing ':'
+                libdir=${libdir:0:${#libdir}-1}
+                if [ -e $libdir/$obj_file ]; then
+                    ln -s $libdir/$obj_file ${PREFIX}/lib/gcc/*/*/
+                    break
+                fi
+            done
+        done
+
+        #
+        # Linux Portability Issue #3: Compiler needs to locate system headers
+        #
+    
+        # Some distros use different system include paths than the ones this gcc binary was built for.
+        # We'll add these to the standard include path by providing a custom "specs file"
+    
+        # First create specs file from existing defaults
+        SPECS_DIR=`echo ${PREFIX}/lib/gcc/*/*`
+        SPECS_FILE=${SPECS_DIR}/specs
+        gcc -dumpspecs > ${SPECS_FILE}
+        
+        # Now add extra include paths to the specs file, one at a time.
+        # (So far we only know of one: from Ubuntu.)
+        EXTRA_SYSTEM_INCLUDE_DIRS="/usr/include/x86_64-linux-gnu"
+        
+        for INCDIR in ${EXTRA_SYSTEM_INCLUDE_DIRS}; do
+            # The following sed command will replace these two lines:
+            # *cpp:
+            # ... yada yada ...
+            #
+            # With these two lines:
+            # *cpp:
+            # ... yada yada ... -I${INCDIR}
+            sed -i ':a;N;$!ba;s|\(*cpp:\n[^\n]*\)|\1 -I'${INCDIR}'|g' ${SPECS_FILE}
```

Comment:
Are you referring to the in-place portion? I haven't seen this with Linux, but this could always be changed to `-i.bak` or some other extension and that should work in all cases. Now, it would probably be a good idea to go back and clean these original files.